app/Http/Controllers/Api/ReportbackController.php

```diff
@@ -23,15 +24,28 @@ public function __construct(ReportbackService $reportbackService)
     }
 
     /**
-     * Store a newly created resource in storage.
-     * @todo upsert reportbacks by default.
+     * Store a newly created resource in storage or
+     * update a reportback if it already exists.
+     *
      * @param  \Illuminate\Http\Request  $request
      * @return \Illuminate\Http\Response
      */
-    public function create(Request $request)
+    public function store(ReportbackRequest $request)
     {
-        $reportback = $this->reportbackService->create($request->all());
+        // @TODO - instead should probably just have a method that gets northstar_id by default from a drupal_id if that is the only thing provided and then use that to find the reportback.
+        $userId = $request['northstar_id'] ? $request['northstar_id'] : $request['drupal_id'];
+        $type = $request['northstar_id'] ? 'northstar_id' : 'drupal_id';
 
-        return $this->item($reportback);
+        $reportback = $this->reportbackService->exists($request['campaign_id'], $request['campaign_run_id'], $userId, $type);
+
+        if (! $reportback) {
+            $reportback = $this->reportbackService->create($request->all());
+
+            return $this->item($reportback);
+        } else {
+            $updatedReportback = $this->reportbackService->update($reportback, $request->all());
+
+            return $this->item($updatedReportback);
```

Comment:
Another nice thing would be to keep track of which branch of the conditional is run (say, with an `$updating` boolean) that can then be passed along as either a 200 or 201 status code to the response method. Here's an [example](https://github.com/DoSomething/northstar/blob/e6255f2fbc5cd162587867adb2457c07bfd81b51/app/Http/Controllers/UserController.php#L123-L125) of a similar kind of thing on Northstar's `POST users/` endpoint.